src/org/rust/lang/core/psi/impl/mixin/RustPatFieldImplMixin.kt

```diff
@@ -0,0 +1,19 @@
+package org.rust.lang.core.psi.impl.mixin
+
+import com.intellij.lang.ASTNode
+import org.rust.lang.core.psi.RustNamedElement
+import org.rust.lang.core.psi.RustPatField
+import org.rust.lang.core.psi.impl.RustNamedElementImpl
+import org.rust.lang.core.psi.util.boundElements
+
+public abstract class RustPatFieldImplMixin(node: ASTNode)  : RustNamedElementImpl(node)
+                                                            , RustPatField {
+
+    override fun getBoundElements(): Collection<RustNamedElement> {
+        val boundInPat = pat?.boundElements
+                            .orEmpty().
+                            toTypedArray()
+
+        return arrayListOf(this, *boundInPat)
```

Comment:
I believe this `this` is wrong. Consider:

```rust
struct S {
    foo: i32
}

match ... {
    S {foo} => /* here name foo is boud */
    S {foo : bar} => /* here foo is unboud, but bar is bound */
}
```

We do have a test case for this: `struct_patterns(1|2).rs`